### PR TITLE
Use snake-case for slash-munged paths in packager

### PR DESCRIPTION
**Pull request by aadi-stripe**
For the newly added "slash-munged" package paths in #5987 , we need to also snake-case the path names in addition to lower-casing, so that `Project_FooBar` transforms to `project/foo_bar`. Previously it was just `project/foobar`.

### Motivation

In the Stripe codebase, we are working on introducing path-based autoloading based on [Zeitwerk](https://github.com/fxn/zeitwerk)'s convention, which is used in Rails codebases.

### Test plan

See included automated tests. Tested on Stripe codebase.

### CODE REVIEW

**Review by bhang-stripe — APPROVED** [review]

**Comment by aadi-stripe** [comment]
Will merge after review from core reviewers, aka @jez or @froydnj

**Review by jez — APPROVED** [review]

**Pull request merged**
